ring/replication_set.go

```diff
@@ -468,6 +468,17 @@ func HasReplicationSetChangedWithoutState(before, after ReplicationSet) bool {
 	})
 }
 
+// Has HasReplicationSetChangedWithoutStateOrAddr returns false if two replications sets
```

Comment:
You don't have to do it as part of this PR but we seem to be accumulating quiet a few different methods that all do roughly the same thing. Maybe we could apply the option pattern here.

e.g.
```go
func HasReplicationSetChanged(before, after ReplicationSet, opt... RingOption) bool { /* ... */ }
```
Called like
```go
if ring.HasReplicationSetChanged(a, b, ring.ExcludeAddress(), ring.ExcludeState()) {
    /* ... */
}
```

Reply:
I had the same thought, but it would mean breaking changes [not only for Mimir](https://github.com/search?q=org%3Agrafana%20HasReplicationSetChanged&type=code). Since dskit doesn't have a formal versioning process we'd need to coordinate changing over (or have two implementations for some time). We also don't have an easy way to signal that breaking change to any external users of dskit (are there any?).

We could add something like

```
func HasReplicationSetChangedFn(opt... ExcludeOption) func (before, after ReplicationSet) bool { /* ... */ }
```

and replace the [usage of `HasReplicationSetChangedWithoutXXXXX`](https://github.com/search?q=org%3Agrafana+HasReplicationSetChangedWithout&type=code), but I don't foresee us needing another flavor of this kind of function in the future (there aren't many other fields left to ignore). If we do end up adding a third kind, though, we should revisit this.